scripts/generate_release_notes.py

```diff
@@ -0,0 +1,263 @@
+        self.get_old_pull_requests()
+        print("Done!")
+
+    def get_new_pull_requests(self):
+        """Get PRs created after the latest release.
+
+        This operation also populates a set of previous release commits.
+        """
+        for pull_request in self.remote_repo.get_pulls(state="closed"):
+            # Stop getting pull requests after previous release tag.
+            if pull_request.title == self.latest_tag_name:
+                # Get previous release commits SHAs.
+                for commit in pull_request.get_commits():
+                    self.previous_commits.add(commit.sha)
+                break
+
+            # Skip closed unmerged PRs.
+            if not pull_request.merged:
+                continue
+
+            if pull_request.number in self.args.exclude:
+                if self.args.verbose:
+                    print(f"Excludeing PR #{pull_request.number} as requested")
+                continue
+
+            if self.args.verbose:
+                messages = [f"Fetching PR #{pull_request.number}"]
+                if pull_request.number in self.args.author_only:
+                    messages.append(
+                        "(keeping PR author as a sole contributor)"
+                    )
+                print(" ".join(messages))
+
+            self.add_pull_request(pull_request)
+
+    def get_old_pull_requests(self):
+        """Get PRs created before the latest release."""
+        pull_request_numbers = set()
+        for commit in self.local_repo.iter_commits():
+            if commit.hexsha in self.previous_commits:
+                break
+
+            try:
+                pull_request_number = re.findall(
+                    r"#(\d{3,})",
+                    commit.message,
+                )[0]
+                pull_request_numbers.add(int(pull_request_number))
+            except IndexError:
+                continue
+
+        # Fetch old PRs metadata only. Skip all known PRs.
+        pull_request_numbers -= set(self.pull_requests.keys())
+        for pull_request_number in pull_request_numbers:
+            if self.args.verbose:
+                messages = [f"Fetching PR #{pull_request_number}"]
+                if pull_request_number in self.args.author_only:
+                    messages.append(
+                        "(keeping PR author as a sole contributor)"
+                    )
+                print(" ".join(messages))
+
+            try:
+                self.add_pull_request(
+                    self.remote_repo.get_pull(pull_request_number)
+                )
+            # 3rd party contributions to forks.
+            except UnknownObjectException:
+                pass
+
+    def print_release_notes(self):
+        """Print generated release notes."""
+        print("")
+        if self.pull_requests:
+            today = date.today()
+            print(
+                HEADER_TEMPLATE.format(
+                    day=today.strftime("%-d"),
```

Comment:
Feature with "-" don't work on Win :(  Maybe it's better to use 
```suggestion
                    day=f"{today.day}",
```

Reply:
Thanks for catching this.

Probably, some basic tests should have been added.